### ISSUE TITLE
Don't dispose ScrollController that pass as parameter

### ISSUE THREAD

**Issue by Daniel-Ioannou** [Dimibe/grouped_list#33]
This packages calls .dispose() on the ScrollController argument it receives.

This means the obvious usage pattern is now wrong:
```
class _MyWidgetState extends State<MyWidget> {
  ScrollController _myController;

  @override
  void initState() {
    super.initState();
    _scrollController = ScrollController()..addListener(_scrollListener);
  }

  @override
  void dispose() {
    // Wrong, because this plugin dispose my own controller and I can not remove listener!
    _scrollController.removeListener(_scrollListener)..
    // Wrong, because this plugin wants me not to dispose my own controller!
    _scrollController.dispose();
    super.dispose();
  }

  @override
  Widget build() {
    return GroupedListView(controller: _scrollController, child: ... );
  }
}
```
This is surprising: If I'm creating the controller and pass it in, it should also be my job to dispose it. This is the way it works everywhere else in Flutter.

**Comment by Dimibe** [comment]
@Daniel-Ioannou The way the package is designed it needs a controller. So if you don't pass one internally one is created. 
This is the trade-off I've made to provide the `controller` option for external use.
The other option was to not provide the `controller` option for external use.

**Comment by Daniel-Ioannou** [comment]
@Dimibe If controller pass as argument it must not dispose. If I'm creating the controller and pass it in, it should also be my job to dispose it. So if a controller don't pass dispose the controller that created internally.

```
@override
void dispose() {
  _controller.removeListener(_scrollListener);
  if (widget.controller == null) {
    _controller.dispose();
  }
  super.dispose();
}
```

**Comment by Dimibe** [comment]
@Daniel-Ioannou Feel free to open a pr.

**Comment by Dimibe** [comment]
@Daniel-Ioannou Thanks, your PR has been merged and is available in version 2.3.1

**Comment by Daniel-Ioannou** [comment]
Thanks @Dimibe for your cooperation.
